File: backend/internal/models/worker.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type WorkerStatus string

const (
	WorkerStatusBusy        WorkerStatus = "busy"
	WorkerStatusPendingUser WorkerStatus = "pending_user"
	WorkerStatusIdle        WorkerStatus = "idle"
	WorkerStatusOffline     WorkerStatus = "offline"
)

type Worker struct {
	WorkerID     uuid.UUID    `db:"worker_id"`
	GroupID      uuid.UUID    `db:"group_id"`
	// GroupName is populated by queries that JOIN with worker_groups.
	GroupName    string       `db:"group_name"`
	Name         string       `db:"name"`
	CallbackURL  string       `db:"callback_url"`
	APIKeyHash   string       `db:"api_key_hash"`
	Workspace    string       `db:"workspace"`
	CLICommand   string       `db:"cli_command"`
	GitRepoURL   string       `db:"git_repo_url"`
	Status       WorkerStatus `db:"status"`
	LastActiveAt *time.Time   `db:"last_active_at"`
	CreatedAt    time.Time    `db:"created_at"`
	DeletedAt    *time.Time   `db:"deleted_at"`
}

File: frontend/src/types/worker.ts
export type WorkerStatus = 'busy' | 'pending_user' | 'idle' | 'offline';
export type CLICommand = 'claude' | 'codex';

export interface Worker {
	worker_id: string;
	group_name: string;
	name: string;
	cli_command: string;
	build_command?: string;
	workspace: string;
	git_repo_url: string;
	instruction_job: string;
	instruction_plan: string;
	instruction_discuss: string;
	map_x: number;
	map_y: number;
	status: WorkerStatus;
	last_active_at?: string;
	created_at: string;
}

export interface CreateWorkerPayload {
	name: string;
	git_repo_url: string;
	group_name: string;
	cli_command: CLICommand;
}

export interface PingResult {
	ok: boolean;
	output: string;
	error?: string;
}

export interface UpdateWorkerPayload {
	cli_command?: CLICommand;
	build_command?: string;
	map_x?: number;
	map_y?: number;
	instruction_job?: string;
	instruction_plan?: string;
	instruction_discuss?: string;
}
